src/ui/components/EventsCalendarTeaser.tsx

```diff
@@ -17,8 +17,8 @@ export function EventsCalendarTeaser(props: Props) {
   const {data: events, loading} = useCalendarEvents();
   const topThreeEvents = events?.slice(0, 3) ?? [];
   return (
-    <SectionTeaserContainer onPress={props.onPress} title="Events Calendar">
-      {loading && !events ? (
+    <SectionTeaserContainer onPress={props.onPress} title="Upcoming events">
+      {loading && !topThreeEvents ? (
```

Comment:
The value of the `topThreeEvents` variable is never falsy. So `!topThreeEvents` will make it false every time and loading will never appear.

```suggestion
      {loading && !events ? (
```

Reply:
I should Have kept a close eye on merging. It is resolved now.